superset/result_set.py

```diff
@@ -48,14 +47,14 @@ def dedup(l: List[str], suffix: str = "__", case_sensitive: bool = True) -> List
     """
     new_l: List[str] = []
     seen: Dict[str, int] = {}
-    for s in l:
-        s_fixed_case = s if case_sensitive else s.lower()
+    for string in l:
```

Comment:
nit: We already know it's a string, I propose calling `item` instead of `string`